astropy/wcs/wcs.py

```diff
@@ -1023,12 +1026,14 @@ def write_dist(num, cpdis):
             hdulist[0].header[f'{d_kw}{num:d}.EXTVER'] = (
                 num, 'Version number of WCSDVARR extension')
             hdulist[0].header[f'{d_kw}{num:d}.NAXES'] = (
-                len(cpdis.data.shape), 'Number of independent variables in distortion function')
+                len(cpdis.data.shape), f'Number of independent variables in {dist} function')
+
+            jth = {1: '1st', 2: '2nd', 3: '3rd'}.get(num, f'{num}th')
```

Comment:
This line also needs to go in the loop and use `i+1`.

Reply:
Oh, yes. Thanks!